datafusion/sql/src/statement.rs

```diff
@@ -710,6 +710,25 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
                     *statement,
                     &mut planner_context,
                 )?;
+
+                //Get inferred data_types from the plan if it is empty in the prepare statement
+                let data_types = match data_types.is_empty() {
```

Comment:
is there a reason to use a match rather than just if/else?

```suggestion
                let data_types = if data_types.is_empty() {
```
...
?

Reply:
No, if/else will do.